### DISCUSSION TITLE
Userflow testing with headless testing

### DISCUSSION THREAD

**Discussion by SivasubramanianV** [GoogleChrome/lighthouse#15344] [Q&A] [answered]
Can we do userflow testing with wdio headless mode?
How does the result of this reflect the real user scenario?
Which is better for reflecting actual web performance?Head or headless

**Comment by adamraine** [comment, marked as answer]
> Can we do userflow testing with wdio headless mode?

We do not officially support wdio but I would expect a `getPuppeteer` for the headless browser to work.

> How does the result of this reflect the real user scenario?
> Which is better for reflecting actual web performance?Head or headless

If you are using the *old* headless mode there can be some differences from standard browser execution. In general I would recommend using the [*new* headless mode](https://developer.chrome.com/articles/new-headless/) which should mirror normal browser execution. I'm not sure which one wdio is using though.

**Reply by SivasubramanianV** [reply]
Thank you Adam. Thats enlightening.
